book/src/chapters/building.md

```diff
@@ -0,0 +1,341 @@
+- Use bundled message definitions for common types
+- Opt-in to ROS 2 integration when needed
+- Pay only for what you use
+
+## Quick Start
+
+Get running in 30 seconds:
+
+```bash
+# Build core library
+cargo build
+
+# Run tests
+cargo test
+
+# Try an example
+cargo run --example z_pubsub
+```
+
+```admonish tip
+The default build requires only Rust and Cargo. No ROS 2, no external dependencies, no configuration files.
+```
+
+## Build Scenarios
+
+```mermaid
+flowchart TD
+    A[Your Project] --> B{Need ROS Messages?}
+    B -->|No| C[Pure Rust<br/>Custom Messages]
+    B -->|Yes| D{Which Messages?}
+    D -->|Common Types| E[Bundled Messages<br/>No ROS Required]
+    D -->|All Types| F[Full ROS Integration<br/>ROS 2 Required]
+
+    C --> C1[cargo build]
+    E --> E1[cargo build -p ros-z-msgs]
+    F --> F1[cargo build --features external_msgs]
+```
+
+### Scenario 1: Pure Rust Development
+
+**Use when:** Prototyping, standalone apps, custom messages only
+
+**Requirements:** Rust + Cargo
+
+```bash
+# Build workspace (ros-z + ros-z-codegen)
+cargo build
+
+# Run tests
+cargo test
+
+# Build example with custom messages
+cargo build -p ros-z --example z_custom_message
+cargo run --example z_custom_message -- --mode status-pub
+```
+
+**What you get:**
+
+- Full ros-z functionality
+- Custom message support
+- Zero external dependencies
+- Fast build times
+
+```admonish info
+Perfect for initial development and MVP builds. Add message packages incrementally as needs grow.
+```
+
+### Scenario 2: Using Bundled Messages
+
+**Use when:** Need standard ROS types, no ROS 2 installation available
+
+**Requirements:** Rust + Cargo (still no ROS 2!)
+
+```bash
+# Build with bundled messages
+cargo build -p ros-z-msgs
+
+# Build examples using standard types
+cargo build -p ros-z --example z_pubsub        # std_msgs
+cargo build -p ros-z --example twist_pub       # geometry_msgs
+cargo build -p ros-z --example laser_scan      # sensor_msgs
+```
+
+**Bundled packages** (via roslibrust):
+
+- `std_msgs` - Strings, integers, arrays
+- `geometry_msgs` - Points, poses, transforms
+- `sensor_msgs` - LaserScan, Image, Imu
+- `nav_msgs` - Path, occupancy grids
+
+```admonish success
+Bundled messages provide ROS 2 compatibility without requiring ROS 2 installation. Perfect for development machines.
+```
+
+### Scenario 3: Full ROS 2 Integration
+
+**Use when:** Need all message types, ROS 2 interoperability required
+
+**Requirements:** Rust + Cargo + ROS 2 installation
+
+```bash
+# Ensure ROS 2 is sourced
+source /opt/ros/jazzy/setup.bash
+
+# Build with external messages
+cargo build -p ros-z-msgs --features external_msgs
+
+# Build RCL bindings
+cargo build -p rcl-z
+
+# Build service examples
+cargo build --example z_srvcli --features external_msgs
+```
+
+**Additional packages:**
+
+- `example_interfaces` - Tutorial services
+- `action_msgs` - Action types
+- Custom packages from your workspace
+
+## Workspace Structure
```

Comment:
The name `Workspace` can be confused with a ROS Workspace. My understanding is that this chapter rather describes this repository organisation, not what would be a ros-z user workspace.